pkg/sparrow/gitlab/gitlab.go

```diff
@@ -0,0 +1,311 @@
+// sparrow
+// (C) 2023, Deutsche Telekom IT GmbH
+//
+// Deutsche Telekom IT GmbH and all other contributors /
+// copyright owners license this file to you under the Apache
+// License, Version 2.0 (the "License"); you may not use this
+// file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package gitlab
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/caas-team/sparrow/internal/logger"
+	"github.com/caas-team/sparrow/pkg/checks"
+)
+
+// Gitlab handles interaction with a gitlab repository containing
+// the global targets for the Sparrow instance
+type Gitlab interface {
+	FetchFiles(ctx context.Context) ([]checks.GlobalTarget, error)
+	PutFile(ctx context.Context, file File) error
+	PostFile(ctx context.Context, file File) error
+}
+
+// Client implements Gitlab
+type Client struct {
+	// the base URL of the gitlab instance
+	baseUrl string
+	// the ID of the project containing the global targets
+	projectID int
+	// the token used to authenticate with the gitlab instance
+	token  string
+	client *http.Client
+}
+
+func New(baseURL, token string, pid int) Gitlab {
+	return &Client{
+		baseUrl:   baseURL,
+		token:     token,
+		projectID: pid,
+		client:    &http.Client{},
+	}
+}
+
+// FetchFiles fetches the files from the global targets repository from the configured gitlab repository
+func (g *Client) FetchFiles(ctx context.Context) ([]checks.GlobalTarget, error) {
+	log := logger.FromContext(ctx)
+	fl, err := g.fetchFileList(ctx)
+	if err != nil {
+		log.Error("Failed to fetch files", "error", err)
+		return nil, err
+	}
+
+	var result []checks.GlobalTarget
+	for _, f := range fl {
+		gl, err := g.fetchFile(ctx, f)
+		if err != nil {
+			log.Error("Failed fetching files", "error", err)
+			return nil, err
+		}
+		result = append(result, gl)
+	}
+	log.Info("Successfully fetched all target files", "files", len(result))
+	return result, nil
+}
+
+// fetchFile fetches the file from the global targets repository from the configured gitlab repository
+func (g *Client) fetchFile(ctx context.Context, f string) (checks.GlobalTarget, error) {
+	log := logger.FromContext(ctx)
```

Comment:
you could add `file` to logger as attribute since you are using it in nearly all logs: `.With("file", f)`